types/v56/types.go

```diff
@@ -494,29 +494,35 @@ type VdcConfiguration struct {
+	Operation        string           `xml:"operation,attr,omitempty"`        // A message describing the operation that is tracked by this task.
+	OperationName    string           `xml:"operationName,attr,omitempty"`    // The short name of the operation that is tracked by this task.
+	ServiceNamespace string           `xml:"serviceNamespace,attr,omitempty"` // Identifier of the service that created the task. It must not start with com.vmware.vcloud and the length must be between 1 and 128 symbols.
+	StartTime        string           `xml:"startTime,attr,omitempty"`        // The date and time the system started executing the task. May not be present if the task has not been executed yet.
+	EndTime          string           `xml:"endTime,attr,omitempty"`          // The date and time that processing of the task was completed. May not be present if the task is still being executed.
+	ExpiryTime       string           `xml:"expiryTime,attr,omitempty"`       // The date and time at which the task resource will be destroyed and no longer available for retrieval. May not be present if the task has not been executed or is still being executed.
+	CancelRequested  bool             `xml:"cancelRequested,attr,omitempty"`  // Whether user has requested this processing to be canceled.
+	Description      string           `xml:"Description,omitempty"`           // Optional description.
+	Details          string           `xml:"Details,omitempty"`               // Detailed message about the task. Also contained by the Owner entity when task status is preRunning.
+	Error            *Error           `xml:"Error,omitempty"`                 // Represents error information from a failed task.
+	Link             *Link            `xml:"Link,omitempty"`                  // A reference to an entity or operation associated with this object.
+	Organization     *Reference       `xml:"Organization,omitempty"`          // The organization to which the User belongs.
+	Owner            *Reference       `xml:"Owner,omitempty"`                 // Reference to the owner of the task. This is typically the object that the task is creating or updating.
+	Progress         int              `xml:"Progress,omitempty"`              // Read-only indicator of task progress as an approximate percentage between 0 and 100. Not available for all tasks.
+	Tasks            *TasksInProgress `xml:"Tasks,omitempty"`                 // A list of queued, running, or recently completed tasks associated with this entity.
+	User             *Reference       `xml:"User,omitempty"`                  // The user who started the task.
+	//
+	// TODO: add the following fields
+	// Params      anyType        The parameters with which this task was started.
+	// Result      ResultType	    An optional element that can be used to hold the result of a task.
+	// VcTaskList  VcTaskListType List of Virtual Center tasks related to this vCD task.
```

Comment:
Why are we adding a TODO, but not the actual fields? :)

Reply:
I added the descriptions because they were missing, but I didn't add more fields because the Task type is used everywhere, and changing it would require more extensive tests than what we have now. In fact, we don't have tests specific to tasks: they are tested together with the entities that they process.